src/main.cpp

```diff
@@ -243,58 +232,60 @@ void find_texture_difference()
 
 	// Load textures
 	glActiveTexture(GL_TEXTURE0);
-	glBindTexture(GL_TEXTURE_2D, context->sceneTexture);
+	glBindTexture(GL_TEXTURE_2D, texture1);
 	glActiveTexture(GL_TEXTURE1);
-	glBindTexture(GL_TEXTURE_2D, context->targetTexture);
+	glBindTexture(GL_TEXTURE_2D, texture2);
 
 	// Get image difference and draw it to whole buffer
 	glBindVertexArray(context->diffVAO);
 	glDrawElements(GL_TRIANGLES, 6, GL_UNSIGNED_INT, 0);
 
 	// Switch to screen output buffer for safety to stop anything else being rendered to the FBO
 	glBindFramebuffer(GL_FRAMEBUFFER, 0);
-	render_to_screen();
+	render_to_screen(context->diffTexture);
 }
 
 void set_target_img(const char *str)
 {
-	if (!context) {
-		init_context();
-	}
 	context->targetTexture = load_texture(str);
 }
 
+
+sceneVertex create_vertex(GLfloat x, GLfloat y, GLfloat z) {
+	sceneVertex vertex; 
+	vertex.x = x; 
+	vertex.y = y;
+	vertex.z = z;
+	return vertex;
+}
```

Comment:
```suggestion
inline sceneVertex create_vertex(GLfloat x, GLfloat y, GLfloat z) {
	sceneVertex vertex; 
	vertex.x = x; 
	vertex.y = y;
	vertex.z = z;
	return vertex;
}
```
Might be worth explicitly marking this as inline? The compiler will probably do it anyway, but always worth hinting it to the compiler :)